src/lib/orionld/serviceRoutines/orionldPostEntityOperationsUpsert.cpp

```diff
@@ -0,0 +1,331 @@
+#include "orionld/context/orionldContextLookup.h"                          // orionldContextLookup
+#include "orionld/context/orionldContextItemLookup.h"                      // orionldContextItemLookup
+#include "orionld/context/orionldContextList.h"                            // orionldContextHead, orionldContextTail
+#include "orionld/context/orionldContextListInsert.h"                      // orionldContextListInsert
+#include "orionld/context/orionldContextPresent.h"                         // orionldContextPresent
+#include "orionld/context/orionldUserContextKeyValuesCheck.h"              // orionldUserContextKeyValuesCheck
+#include "orionld/context/orionldUriExpand.h"                              // orionldUriExpand
+#include "orionld/serviceRoutines/orionldPostEntityOperationsUpsert.h"     // Own Interface
+#include "orionld/mongoCppLegacy/mongoCppLegacyEntityOperationsUpsert.h"   // mongoCppLegacyEntityOperationsUpsert
+#include "orionld/common/orionldEntitiyPayloadCheck.h"                     // payloadCheck
+
+
+// ----------------------------------------------------------------------------
+//
+// orionldPartialUpdateResponseCreateBatch -
+//
+void orionldPartialUpdateResponseCreateBatch(ConnectionInfo* ciP)
+{
+  //
+  // Rob the incoming Request Tree - performance to be won!
+  //
+  orionldState.responseTree = orionldState.requestTree;
+  orionldState.requestTree  = NULL;
+
+  //
+  // For all attrs in orionldState.responseTree, remove those that are found in orionldState.errorAttributeArrayP.
+  // Remember, the format of orionldState.errorAttributeArrayP is:
+  //
+  //   |attrName|attrName|attrName|...
+  //
+
+  KjNode* attrNodeP = orionldState.responseTree->value.firstChildP;
+
+  while (attrNodeP != NULL)
+  {
+    char*   match;
+    KjNode* next = attrNodeP->next;
+    bool    moved = false;
+
+    if ((match = strstr(orionldState.errorAttributeArrayP, attrNodeP->name)) != NULL)
+    {
+      if ((match[-1] == '|') && (match[strlen(attrNodeP->name)] == '|'))
+      {
+        kjChildRemove(orionldState.responseTree, attrNodeP);
+        attrNodeP = next;
+        moved = true;
+      }
+    }
+
+    if (moved == false)
+      attrNodeP = attrNodeP->next;
+  }
+}
+
+
+
+// -----------------------------------------------------------------------------
+//
+// kjTreeToContextElementBatch -
+//
+bool kjTreeToContextElementAttributes
+(
+  ConnectionInfo* ciP,
+  KjNode* treeP,
+  KjNode* createdAtPP,
+  KjNode* modifiedAtPP,
+  ContextElement* ceP
+)
+{
+  // Iterate over the object, to get all attributes
+  while (treeP != NULL)
+  {
+    KjNode*           attrTypeNodeP = NULL;
+    ContextAttribute* caP           = new ContextAttribute();
+
+    if
+    (
+      treeP != createdAtPP              &&
+      treeP != modifiedAtPP             &&
+      strcmp(treeP->name, "id") != 0    &&
+      strcmp(treeP->name, "type") != 0
+    )
```

Comment:
Looks a little funny.
Better like this (also adding parenthesis for each part-condition, to avoid mistakes):
```
if￼ ((treeP != createdAtPP)               &&
    (treeP  != modifiedAtPP)              &&
    (strcmp(treeP->name, "id")   != 0)    &&
    (strcmp(treeP->name, "type") != 0))
{
   ...
```

Reply:
FIXED